### DISCUSSION TITLE
Proper API for result table alterations

### DISCUSSION THREAD

**Discussion by BeyondEvil** [pytest-dev/pytest-html#648] [Ideas]
Currently (as of `v4.0`) the API for the different `pytest_html_results_table_*` hook is hacky at best.

I would like to identify what type of changes are made, and then build a proper API around that.

Here's a list of things that I've discovered so far:

- Adding column(s)
  - Making added column(s) sortable
- Removing columns(s) 
- Removing rows

I should be fairly easy to construct a concise API for these actions.

**Comment by innicoder** [comment]
Agreed

**Reply by BeyondEvil** [reply]
I'm working on a solution for this.

It's a bit of a mess, but that's only because I want to keep it backwards-compat for v4. In v5, we'll remove all the mess and keep a clean API.

Would you be interested in testing it?

**Comment by innicoder** [comment]
Yes. Let me know what you need from me.

More also on this thread: https://github.com/pytest-dev/pytest-html/issues/670
